runtime/drivers/salesforce/bulk.go

```diff
@@ -0,0 +1,292 @@
+package salesforce
```

Comment:
TBH I still don't think its the right place to have the logic contained in the file here. I think the library should encapsulate the logic here and should provide APIs to start the job and retrieve results(may be via some iterator). You can take a look at the bigquery SDK: https://github.com/googleapis/google-cloud-go/blob/0020ab23b0d1344cf03b2bfa5e2afe947b70240a/bigquery/query.go#L373

It provides APIs to create a client(similar to a force session here), create and start a job via `client.Query.Read` method(similar to `makeBulkJob` and `startJob` here) and exposes an iterator to query results(similar to `fetchBatchResult`) here.

Reply:
Yeah I agree that moving this to a library requires lot more thinking on multiple use cases.

I also happen to notice that salesforce has a newer version of the Bulk API which automatically performs chunking and provides result from a single endpoint similar to what I suggested to be done upstream.
Could you please evaluate that? Salesforce also recommends to use Bulk API 2.0 instead of Bulk API for simpler integration.

Reply:
I expect there will be new problems to troubleshoot moving to Bulk API 2.0.  The docs suggest that you can get [query timeouts](https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/bulk_api_2_0_troubleshoot_query_timeouts.htm), and may get [timeouts if you choose the wrong number of records to retrieve at a time](https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/query_get_job_results.htm).

I'm more comfortable using the original Bulk API at this point as we've already identified the rough edges and have been successfully using it for a decade.